fleet-be/src/main/java/com/everis/training/fleet/business/fleet/control/VehicleController.java

```diff
@@ -22,27 +23,45 @@
         return repo.getAllFreeVehicles();
     }
 
-    public void add(Vehicle vehicle) {
-        if (vehicle.getVin().length()==17){
-            repo.saveVehicle(vehicle);
+    public void delete(String vin) throws Exception{
+        try {
+            vinCheck(vin);
+            repo.deleteVehicle(vin);
+        } catch (Exception e) {
+            e.printStackTrace();
         }
     }
 
-    public void delete(String vin) {
-        if (vin.length() == 17){
-            repo.deleteVehicle(vin);
+    public void reserveCar(Integer id, String vin) throws Exception{
+        try {
+            vinCheck(vin);
+            repo.reserveCar(id, vin);
+        } catch (Exception e) {
+            e.printStackTrace();
         }
-
     }
 
-    public void update(Vehicle vehicle){
-        if (vehicle.getVin().length()==17){
-            repo.updateVehicle(vehicle);
+    public void add(Vehicle vehicle) throws Exception {
+        try {
+            vinCheck(vehicle.getVin());
+            repo.addVehicle(vehicle);
+        } catch (Exception e) {
+            throw e;
         }
     }
 
     public Vehicle findVehicle(String vin){
         return repo.findVehicle(vin);
     }
 
```

Comment:
Please, use always as possible the more specific Exception in the method signature, that also helps to understand better what the method does.
The method name should be: checkVin (first part is the action)